app/graphql/resolvers/create_offer_response_resolver.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class CreateOfferResponseResolver < BaseResolver
+  def valid?
+    return true if user?
+
+    bad_argument_error =
+      GraphQL::ExecutionError.new("Can't access createConsignmentOfferResponse")
+    @error = bad_argument_error
+    false
+  end
+
+  def run
+    offer = Offer.find(offer_id)
+    matching_user =
+      offer.submission&.user&.gravity_user_id == @context[:current_user]
```

Comment:
Ensure that the offer they're responding to is indeed their own.

Reply:
**question [non-blocking]:** check validity of intent sooner?

What happens if one tries to respond to an offer where they are not the gravity user? I wonder if this check belongs in the `valid?` method? 🤔 

*format via [Conventional Comments](https://conventionalcomments.org/)*

Reply:
That check is just below!

Actually perhaps a question for you-- in the resolvers, I see a mix of approaches (some validations in `valid?` and others within the resolvers themselves).

I tried to keep the `valid?` methods fairly lean and based off of attributes that we get "for free" (i.e. whether you're an admin, user, partner, etc.) and to leave the more complex checks (for ownership, etc.) to the resolver method.

Many ways to approach this, curious if you have a strong opinion?

Reply:
Yeah I just read through the resolvers and you're right that there are various approaches. I think we are aligned on the line between `valid?` doing some sanity checking and the `run` method raising errors. Where I land on this is that `valid?` should probably stay out of the database. So we can check things like whether the token includes the admin flag or if the combination of query args are allowed, whatever. But checking Rails-level validations or doing something that hits the db is probably out of scope for the valid method.

So given that, I retract my question, haha. What you have here looks wonderful! 🙌